### ISSUE TITLE
Upgrade 3.0 bug? The JWT string must have two dots

### ISSUE THREAD

**Issue by benr77** [Drenso/symfony-oidc#57]
I have upgraded a previously working setup to v3.0 and now I cannot login. The `login_check` endpoint 500 errors with `The JWT string must have two dots`. 

I don't believe any of the upgrade notes affected my setup, my SSO provider (Zitadel) was working fine before the upgrade, so presuming tokens are the same. (i.e. no code has been changed, just the upgrade of the package to 3.0)

Is this because the v3.0 utilises the `lcobucci/jwt` package which is applying stricter validation perhaps?

```
Lcobucci\JWT\Token\InvalidTokenStructure:
The JWT string must have two dots

  at vendor/lcobucci/jwt/src/Token/InvalidTokenStructure.php:13
  at Lcobucci\JWT\Token\InvalidTokenStructure::missingOrNotEnoughSeparators()
     (vendor/lcobucci/jwt/src/Token/Parser.php:65)
```

Thank you

**Comment by bobvandevijver** [comment]
Well, a JWT must indeed have two dots (otherwise it is missing a crucial part), so the question is why that isn't the case with your provider. This does make me curious as to why it is working with v2 for you, so maybe I missed a step somewhere. The stack trace you've provided unfortunately isn't complete, so I do not know which token is responsible for the issue.

I do not have access to an Zitadel SSO provider (and even if I would, it might depend on your settings), but can you check the token format that is returned by Zitadel?

**Comment by benr77** [comment]
I will follow up with Zitadel and come back to you. Thanks

**Comment by benr77** [comment]
```
/vendor/lcobucci/jwt/src/Token/InvalidTokenStructure.php in Lcobucci\JWT\Token\InvalidTokenStructure::missingOrNotEnoughSeparators at line 13

/vendor/lcobucci/jwt/src/Token/Parser.php in Lcobucci\JWT\Token\Parser::splitJwt at line 65

/vendor/lcobucci/jwt/src/Token/Parser.php in Lcobucci\JWT\Token\Parser::parse at line 28

/vendor/drenso/symfony-oidc-bundle/src/OidcJwtHelper.php in Drenso\OidcBundle\OidcJwtHelper::parseToken at line 66

/vendor/drenso/symfony-oidc-bundle/src/OidcJwtHelper.php in Drenso\OidcBundle\OidcJwtHelper::verifyTokens at line 89

/vendor/drenso/symfony-oidc-bundle/src/OidcClient.php in Drenso\OidcBundle\OidcClient::verifyTokens at line 445

/vendor/drenso/symfony-oidc-bundle/src/OidcClient.php in Drenso\OidcBundle\OidcClient::authenticate at line 89

/vendor/drenso/symfony-oidc-bundle/src/Security/OidcAuthenticator.php in Drenso\OidcBundle\Security\OidcAuthenticator::authenticate at line 63

/vendor/symfony/security-http/Authentication/AuthenticatorManager.php in Symfony\Component\Security\Http\Authentication\AuthenticatorManager::executeAuthenticator at line 176

/vendor/symfony/security-http/Authentication/AuthenticatorManager.php in Symfony\Component\Security\Http\Authentication\AuthenticatorManager::executeAuthenticators at line 158

/vendor/symfony/security-http/Authentication/AuthenticatorManager.php in Symfony\Component\Security\Http\Authentication\AuthenticatorManager::authenticateRequest at line 140

/vendor/symfony/security-http/Firewall/AuthenticatorManagerListener.php in Symfony\Component\Security\Http\Firewall\AuthenticatorManagerListener::authenticate at line 40

/vendor/symfony/security-bundle/Security/LazyFirewallContext.php in Symfony\Bundle\SecurityBundle\Security\LazyFirewallContext::__invoke at line 60

/vendor/symfony/security-http/Firewall.php in Symfony\Component\Security\Http\Firewall::callListeners at line 128

/vendor/symfony/security-http/Firewall.php in Symfony\Component\Security\Http\Firewall::onKernelRequest at line 95

/vendor/symfony/event-dispatcher/EventDispatcher.php in Symfony\Component\EventDispatcher\EventDispatcher::Symfony\Component\EventDispatcher\{closure} at line 246

/vendor/symfony/event-dispatcher/EventDispatcher.php in Symfony\Component\EventDispatcher\EventDispatcher::callListeners at line 206

/vendor/symfony/event-dispatcher/EventDispatcher.php in Symfony\Component\EventDispatcher\EventDispatcher::dispatch at line 56

/vendor/symfony/http-kernel/HttpKernel.php in Symfony\Component\HttpKernel\HttpKernel::handleRaw at line 154

/vendor/symfony/http-kernel/HttpKernel.php in Symfony\Component\HttpKernel\HttpKernel::handle at line 76

/vendor/symfony/http-kernel/Kernel.php in Symfony\Component\HttpKernel\Kernel::handle at line 185

/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php in Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner::run at line 35

/vendor/autoload_runtime.php in require_once at line 29

```

**Comment by bobvandevijver** [comment]
From the trace it appears that the ID token is a valid JWT, but the access token isn't. Now I remember what your issue is: the ID token is explicitly specified as JWT, but the access token isn't making it up to the provider to choose to use something. Zitadel seems to have choosen to not use a JWT for the access token. 

With v3 I added JWT validation for the access token as well, which works fine for my providers as they do use a JWT for the both tokens, but breaks for you. I have have now made access token validation optional, and released that as v3.0.1. I am assuming that this fixes the issue for you.

**Comment by benr77** [comment]
Yes, this appears to have sorted it completely. Thanks so much for your quick responses the quick fix!
